File: C++/compile_test.cpp
#include <iostream>
#include <vector>
#include <string>

int fibonacci(int n) {
    if (n <= 1) return n;
    return fibonacci(n - 1) + fibonacci(n - 2);
}

double matrix_multiply(double a[3][3], double b[3][3]) {
    double c[3][3] = {0};
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            for (int k = 0; k < 3; k++) {
                c[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    double sum = 0;
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            sum += c[i][j];
        }
    }
    return sum;
}

int main() {
    volatile int result = 0;
    for (int i = 0; i < 30; i++) {
        result += fibonacci(i);
    }
    
    double a[3][3], b[3][3];
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            a[i][j] = i * 3.0 + j + 1.0;
            b[i][j] = j * 3.0 + i + 1.0;
        }
    }
    
    volatile double sum = 0;
    for (int i = 0; i < 1000000; i++) {
        sum += matrix_multiply(a, b);
    }
    
    return result + (int)sum;
}

File: Java/CompileTest.java
import java.util.ArrayList;
import java.util.List;

public class CompileTest {
    static int fibonacci(int n) {
        if (n <= 1) return n;
        return fibonacci(n - 1) + fibonacci(n - 2);
    }
    
    static double matrixMultiply(double[][] a, double[][] b) {
        int n = a.length;
        double[][] c = new double[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    c[i][j] += a[i][k] * b[k][j];
                }
            }
        }
        double sum = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                sum += c[i][j];
            }
        }
        return sum;
    }
    
    public static void main(String[] args) {
        int result = 0;
        for (int i = 0; i < 30; i++) {
            result += fibonacci(i);
        }
        
        int n = 3;
        double[][] a = new double[n][n];
        double[][] b = new double[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                a[i][j] = i * 3.0 + j + 1.0;
                b[i][j] = j * 3.0 + i + 1.0;
            }
        }
        
        double sum = 0;
        for (int i = 0; i < 1000000; i++) {
            sum += matrixMultiply(a, b);
        }
        
        System.out.println("Result: " + (result + (int)sum));
    }
}
